torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -121,6 +122,8 @@ Reducer::Reducer(
                 torch::make_unique<torch::autograd::utils::LambdaPostHook>(
                     [=](const torch::autograd::variable_list& outputs,
                         const torch::autograd::variable_list& /* unused */) {
+                      this->rpc_context_.set(
+                          ThreadLocalDistAutogradContext().context_weak_ptr);
```

Comment:
Probably should add a static method to this class to get the thread local instead of creating a default instance like this.